hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestFSORepairTool.java

```diff
@@ -0,0 +1,582 @@
+      throws Exception {
+
+    Map<String, Integer> pendingDeleteDirCounts = new HashMap<>();
+
+    // Check deleted directory table.
+    OzoneManager leader = cluster.getOMLeader();
+    Table<String, OmKeyInfo> deletedDirTable =
+        leader.getMetadataManager().getDeletedDirTable();
+    try (TableIterator<String, ?
+        extends Table.KeyValue<String, OmKeyInfo>> iterator =
+            deletedDirTable.iterator()) {
+      while (iterator.hasNext()) {
+        Table.KeyValue<String, OmKeyInfo> entry = iterator.next();
+        String key = entry.getKey();
+        OmKeyInfo value = entry.getValue();
+
+        String dirName = key.split("/")[4];
+        LOG.info("In deletedDirTable, extracting directory name {}  from DB " +
+            "key {}", dirName, key);
+
+        // Check that the correct dir info was added.
+        // FSO delete path will fill in the whole path to the key in the
+        // proto when it is deleted. Once the tree is disconnected that can't
+        // be done, so just make sure the dirName contained in the key name
+        // somewhere.
+        Assertions.assertTrue(value.getKeyName().contains(dirName));
+
+        int count = pendingDeleteDirCounts.getOrDefault(dirName, 0);
+        pendingDeleteDirCounts.put(dirName, count + 1);
+      }
+    }
+
+    // 1 directory is disconnected in the tree. dir1 was totally deleted so
+    // the repair tool will not see it.
+    Assertions.assertEquals(1, pendingDeleteDirCounts.size());
+    Assertions.assertEquals(numWrites, pendingDeleteDirCounts.get("dir2"));
+
+    // Check that disconnected files were put in deleting tables.
+    Map<String, Integer> pendingDeleteFileCounts = new HashMap<>();
+
+    Table<String, RepeatedOmKeyInfo> deletedFileTable =
+        leader.getMetadataManager().getDeletedTable();
+    try (TableIterator<String, ?
+        extends Table.KeyValue<String, RepeatedOmKeyInfo>> iterator =
+            deletedFileTable.iterator()) {
+      while (iterator.hasNext()) {
+        Table.KeyValue<String, RepeatedOmKeyInfo> entry = iterator.next();
+        String key = entry.getKey();
+        RepeatedOmKeyInfo value = entry.getValue();
+
+        String[] keyParts = key.split("/");
+        String fileName = keyParts[keyParts.length - 1];
+
+        LOG.info("In deletedTable, extracting file name {}  from DB " +
+            "key {}", fileName, key);
+
+        for (OmKeyInfo fileInfo: value.getOmKeyInfoList()) {
+          // Check that the correct file info was added.
+          Assertions.assertTrue(fileInfo.getKeyName().contains(fileName));
+
+          int count = pendingDeleteFileCounts.getOrDefault(fileName, 0);
+          pendingDeleteFileCounts.put(fileName, count + 1);
+        }
+      }
+    }
+
+    // 3 files are disconnected in the tree.
+    // TODO: dir2 ended up in here with count = 1. file3 also had count=1
+    //  Likely that the dir2/file3 entry got split in two.
```

Comment:
Looks like there's some old TODOs and commented out code I left behind, including a hardcoded path to a testing DB I was using. Just read through and see if you can make them work for the general case. Let me know if you have questions and I can try to remember what I was thinking 😄

Reply:
Hm yeah, I also haven't been able to figure out what the original test would like to test, so I removed it. Maybe we could add new test whenever required! Thanks~